common/scala/src/main/scala/org/apache/openwhisk/core/entity/WhiskActivation.scala

```diff
@@ -124,6 +124,11 @@ case class WhiskActivation(namespace: EntityPath,
   }
   def withoutLogs = copy(logs = ActivationLogs()).revision[WhiskActivation](rev)
   def withLogs(logs: ActivationLogs) = copy(logs = logs).revision[WhiskActivation](rev)
+
+  def isTimedoutActivation = {
+    annotations.getAs[Boolean](WhiskActivation.timeoutAnnotation).getOrElse(false)
+  }
```

Comment:
This annotation is only added to "invoker" activations - not for compositions which are handled in the controller. Should it be added there also?

Reply:
@rabbah I second your thoughts to to make the time-out annotation consistently available  in the controller. I especially like the added value making the timeout case more visible to the user. Since the underlying problem that triggered this PR is solved with these invoker side changes,  I volunteer to take the controller changes to my list and address it in a subsequent PR.

Reply:
@mhenke1 thanks a lot for volunteering - would you please open a tracking issue for this topic?

I would also highly appreciate to move the annotation work out of this PR to get it done. This PR fixes a problem that should be addressed sooner than later.

@rabbah do you agree with this approach?

Reply:
@sven-lange-last I have opened #4310 to address the controller side work

Reply:
> @rabbah do you agree with this approach?

💯, agree.